src/main/java/com/fasterxml/jackson/module/jsonSchema/JsonSchema.java

```diff
@@ -441,6 +477,9 @@ public boolean isValueTypeSchema() {
 
 	public void set$schema(String $schema) {
 		this.$schema = $schema;
+        if (version == null) {
+            this.version = JsonSchemaVersion.fromSchemaString($schema).orElse(null);
```

Comment:
Why doesn't method just return `null`?

Reply:
The problem is that version isn't part of the JsonSchema so it is annotated with JsonIgnore. So when deserializing the version isn't set. So what this is doing is deriving the version from the schema. Thoughts on a better way to do this?  Perhaps with a custom deserializer?

Reply:
fromSchemaString returns an optional so that you can set it to a default value if you wanted.

IE
JsonSchemaVersion.fromSchemaString($schema).orElse(JsonSchemaVersion.DRAFT_V4);
OR in the future maybe:
JsonSchemaVersion.fromSchemaString($schema).orElse(JsonSchemaVersion.DEFAULT);  where DEFAULT could be the primarily or most recent supported version